layouts/DocsPage/Versions.tsx

```diff
@@ -2,30 +2,90 @@ import cn from "classnames";
 import { useRouter } from "next/router";
 import { useCallback, useEffect, useMemo, useState } from "react";
 import { Dropdown } from "components/Dropdown";
-import type { VersionsInfo } from "./types";
+import type { VersionsInfo, VersionsDropdown } from "./types";
 import styles from "./Versions.module.css";
 
-const renderVersion = (version: string) => `Version ${version}`;
+// renders strikethrough for deprecated versions
+const RenderVersion = (version: VersionsDropdown) => {
+  if (version.deprecated) return <s>Version {version.value}</s>;
+
+  if (version.value === "Older Versions") return version.value;
+  else return `Version ${version.value}`;
+};
+
+// renders the default box selection
+const pickOption = (options: VersionsDropdown[], id: string) =>
+  options.find(({ value }) => value === id);
+
+// assigns keys and values based on the value prop
+const pickId = ({ value }: VersionsDropdown) => value;
 
 const Versions = ({
   current,
   available,
   disabled,
   className,
   getNewVersionPath,
+  latest,
 }: VersionsInfo) => {
   const router = useRouter();
   const [currentItem, setCurrentItem] = useState<string>(current);
-  const versions = useMemo(() => [...available].reverse(), [available]);
 
+  const latestNumber: number = Math.floor(+latest);
+  const validVersion = useCallback(
+    (num: number) => {
+      return num >= latestNumber - 2 ? true : false;
+    },
+    [latestNumber]
+  );
+
+  const versions = useMemo(() => {
+    //creates list of versions ultimately from config.json
+    const versionNames = [...available].reverse();
+
+    //assigns versions a deprecated status: boolean
+    const versionsList = versionNames.map((version) => {
+      const versionNumber: number = +version;
+
+      const versionInfo: VersionsDropdown = {
+        value: version,
+        deprecated: !validVersion(versionNumber),
+      };
+      return versionInfo;
+    });
+
+    //adds an Older Versions element
+    versionsList.push({
+      value: "Older Versions",
+      deprecated: false,
+    });
+    return versionsList;
+  }, [available, validVersion]);
+
+  // only fires when dropdown selection is changed
   const navigateToVersion = useCallback(
-    (version: string) => {
-      const href = getNewVersionPath(version);
+    (option: string) => {
```

Comment:
[Up above](https://github.com/gravitational/docs/pull/55/files#diff-5349a04dbe5b2f520f395199a130bdd2a5eba0d8b925efeb2a26e007bb6eb65eR22) it says `onChange` takes an an argument that can be `string | VersionsDropdown`, but here it says the argument's type is just `string`. Which definition is incorrect - this one or the one above?

Reply:
The underlying `ListboxInput` takes in a handler where input is a `string`, so it seems the one above is incorrect.

Reply:
In strict mode, this is a problem and gives this error:
```
Type '(option: string) => void' is not assignable to type '(selected: string | VersionsDropdown) => void'.
  Types of parameters 'option' and 'selected' are incompatible.
    Type 'string | VersionsDropdown' is not assignable to type 'string'.
      Type 'VersionsDropdown' is not assignable to type 'string'.ts(2322)
```

Reply:
good catch, forgot to update the `onChange` type